scipy/interpolate/tests/test_interpolate.py

```diff
@@ -1341,6 +1341,12 @@ def test_yi_trailing_dims(self):
         pp = BPoly.from_derivatives(xi, yi)
         assert_equal(pp.c.shape, (2*k, m, 6, 7, 8))
 
+    def test_gh_5430(self):
+        orders = np.int64(1)
```

Comment:
As a minimum, this should also test `np.int32`. For instance, the test as written does not fail for me locally, but does fail for np.int32. What matters AFAICS is that the integer is non-native (int32 on 64 bit system or vice versa).